apps/web/src/pages/Pools/EditPoolPage/components/ScreeningQuestions.tsx

```diff
@@ -0,0 +1,356 @@
+    <div data-h2-margin="base(-x.5, 0, x1, 0)" data-h2-text-align="base(right)">
+      <WordCounter
+        text={value[index].question[locale] || ""}
+        wordLimit={TEXT_AREA_MAX_WORDS}
+      />
+    </div>
+  );
+};
+
+const maxWordValidator = (value: string, intl: IntlShape) => {
+  return (
+    countNumberOfWords(value) <= TEXT_AREA_MAX_WORDS ||
+    intl.formatMessage(errorMessages.overWordLimit, {
+      value: TEXT_AREA_MAX_WORDS,
+    })
+  );
+};
+
+interface ScreeningQuestionsProps {
+  poolAdvertisement: PoolAdvertisement;
+  sectionMetadata: EditPoolSectionMetadata;
+  onSave: (submitData: ScreeningQuestionsSubmitData) => void;
+}
+
+const ScreeningQuestions = ({
+  poolAdvertisement,
+  sectionMetadata,
+  onSave,
+}: ScreeningQuestionsProps) => {
+  const intl = useIntl();
+  const [removed, setRemoved] = React.useState<Array<string>>([]);
+  const { isSubmitting } = useEditPoolContext();
+
+  const dataToFormValues = (initialData: PoolAdvertisement): FormValues => ({
+    questions:
+      initialData?.screeningQuestions
+        ?.filter(notEmpty)
+        .map(({ id, question }) => ({
+          id: id || "new",
+          question: {
+            en: question?.en || "",
+            fr: question?.fr || "",
+          },
+        })) || [],
+  });
+
+  const methods = useForm<FormValues>({
+    defaultValues: dataToFormValues(poolAdvertisement),
+  });
+  const { handleSubmit, control } = methods;
+  const { remove, move, append, fields } = useFieldArray({
+    control,
+    name: "questions",
+  });
+
+  const handleRemove = (index: number) => {
+    const toBeRemoved = fields[index];
+    setRemoved((currentlyRemoved) => {
+      if (!currentlyRemoved.includes(toBeRemoved.id)) {
+        return [...currentlyRemoved, toBeRemoved.id];
+      }
+
+      return currentlyRemoved;
+    });
+
+    remove(index);
+  };
+
+  const handleSave = (formValues: FormValues) => {
+    const create: Array<CreateScreeningQuestionInput> = [];
+    const update: Array<UpdateScreeningQuestionInput> = [];
+    const toBeDeleted = poolAdvertisement.screeningQuestions
+      ?.filter((existingQuestion) => {
+        return !formValues.questions?.some(
+          (question) =>
+            question.id === existingQuestion?.id && question.id !== "new",
+        );
+      })
+      .filter(notEmpty)
+      .map((q) => q.id);
+    formValues.questions?.forEach(({ id, question }, index) => {
+      const sortOrder = index + 1;
+      if (!id || id === "new") {
+        create.push({
+          question,
+          sortOrder,
+        });
+      } else {
+        update.push({
+          id,
+          question,
+          sortOrder,
+        });
+      }
+    });
+
+    onSave({
+      screeningQuestions: {
+        update,
+        create,
+        delete: toBeDeleted,
+      },
+    });
+    methods.reset(formValues, {
+      keepDirty: false,
+    });
+    setRemoved([]);
+  };
+
+  // disabled unless status is draft
+  const formDisabled =
```

Comment:
It seems like I can still interact with the repeater when the form is published.  Of course, nothing is being saved in the db but it's weird to be able to reorder and delete questions in the form after publishing.

Reply:
Oh, that totally makes sense. Thanks for highlighting that! Fixed in [disable repeater when pool is published](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6303/commits/8f1a2c67733803cde206a074c3d722dd1c9745ed)